### ISSUE TITLE
Incorrect DateFormat when using an uncommon English locale

### ISSUE THREAD

**Issue by digitalby** [dart-lang/intl#391]
Precondition: iOS with device language set to English (UK) and the region set to Russia, which results in an `en_RU` locale. 

When using `DateFormat`, the date/time is not formatted as expected (using DD.MM.YYYY HH:MM as it is with the system's `DateFormatter`), instead producing an `en_US` formatted string with 12h formatting and MM/DD/YYYY format, which looks confusing to the end-user. 

This happens even when explicitly setting the `DateFormat` locale using a string literal. 

When using `en_GB`, for example, the string is formatted a bit more like expected (Probably because it's a locale that Intl "knows" how to use. )

Would like a clarification if this is a bug, and if it's planned to be fixed; and if there are any possible workarounds currently available (The issue in question is pretty severe and disrupting to my current project.)

**Comment by digitalby** [comment]
I know it's been a year, but just maybe I can get a status update?

**Comment by alan-knight** [comment]
If a locale does not exist in the map of locales (like en_RU) then there is
fallback logic which attempts to find another locale to use, and in this
case it will fall back to "en", which is effectively the same as en_US. If
you give the DateFormat a locale that it knows about (e.g. presumably
ru_RU) then it will probably get better results. Or you can add a locale to
the map, there are examples around here somewhere. I think you might have
to do that if you want the Russian locale formatting, but English
characters on names.

**Comment by digitalby** [comment]
`intl`'s fallback logic is quite flaw in this regard, don't you think? Why mix _regional_ preferences and _language_ preferences? I never had this problem with the native `DateFormatter` as the system automatically picked up the desired date format (DD.MM.YYYY).

I was implementing a date/time picker for a highly internationalized Flutter app when I discovered that basically the entire app's UX would severely suffer (any person outside of US would agree on suddenly seeing a MM.DD.YYYY _just_ in one app when it works as expected everywhere else). Writing a proper workaround would be unreasonable (especially if accounted for all edge cases that arise).

**Comment by alan-knight** [comment]
Yes, the fallback logic is trivial, and very weak. It says so in the
documentation. A good one is very difficult/impossible, and involves
locales the Dart Intl system can't even represent at the moment. There is a
more sophisticated Locale class inside there somewhere, unless it's been
removed. But trying to figure out what you ought to use for a locale that
wasn't provided is always a poor substitute for providing the locale
information.
